src/Hintman/Suggest/TrailingNewline.hs

```diff
@@ -0,0 +1,19 @@
+module Hintman.Suggest.TrailingNewline
+       ( suggestNoTrailingNewline
+       ) where
+
+import Hintman (LineNumber)
+
+import qualified Data.Text as T
+
+
+suggestNoTrailingNewline :: (FilePath, Text) -> Maybe (FilePath, LineNumber, Text)
+suggestNoTrailingNewline (path, body) = (generateResult . reverse . dropWhile isNewline . reverse) splitBody
+    where
+        splitBody :: [Text]
+        splitBody = T.lines body
+        generateResult :: [Text] -> Maybe (FilePath, LineNumber, Text)
+        generateResult newBody | length newBody == length splitBody = Nothing
```

Comment:
Suggest the guards on the next line instead
```
generateResult newBody
  | ..
  | otherwise = ..```

Reply:
I think a run of `stylish-haskell` would be warranted at the end of this pr

Reply:
:+1: didn't realize the yaml was already present, I'll run it before the next commit